operations/helm/charts/mimir-distributed/templates/mimir-config.yaml

```
@@ -1,10 +1,20 @@
 {{- if not .Values.useExternalConfig }}
 apiVersion: v1
+{{- if eq .Values.configStorageType "Secret" }}
 kind: Secret
+{{- else }}
+kind: ConfigMap
+{{- end }}
 metadata:
   name: {{ tpl .Values.externalConfigSecretName . }}
   labels:
     {{- include "mimir.labels" (dict "ctx" .) | nindent 4 }}
+{{- if eq .Values.configStorageType "Secret" }}
 data:
   mimir.yaml: {{ include "mimir.calculatedConfig" . | b64enc }}
+{{- else if eq .Values.configStorageType "ConfigMap" }}
```

I wonder if this should just be `else` like the conditional for the `kind`

---

Yes, I did add a validation as well.

```suggestion
{{- else }}
```